ui/lib/kubernetes/addon/components/page/roles.js

```diff
@@ -9,24 +9,64 @@ import { action } from '@ember/object';
 import { getOwner } from '@ember/application';
 import errorMessage from 'vault/utils/error-message';
 import { tracked } from '@glimmer/tracking';
+import keys from 'core/utils/key-codes';
 
 /**
  * @module Roles
- * RolesPage component is a child component to show list of roles
+ * RolesPage component is a child component to show list of roles.
+ * It also handles the filtering actions of roles.
  *
  * @param {array} roles - array of roles
  * @param {boolean} promptConfig - whether or not to display config cta
- * @param {array} pageFilter - array of filtered roles
+ * @param {string} filterValue - value of queryParam pageFilter
  * @param {array} breadcrumbs - breadcrumbs as an array of objects that contain label and route
  */
 export default class RolesPageComponent extends Component {
   @service flashMessages;
+  @service router;
+  @tracked query;
   @tracked roleToDelete = null;
 
+  constructor() {
+    super(...arguments);
+    this.query = this.args.filterValue;
+  }
+
   get mountPoint() {
     return getOwner(this).mountPoint;
   }
 
+  navigate(pageFilter) {
+    const route = `${this.mountPoint}.roles.index`;
+    const args = [route];
+
+    args.push({
+      queryParams: {
+        pageFilter: pageFilter ? pageFilter : null,
+      },
+    });
+    this.router.transitionTo(...args);
```

Comment:
i think using `push` here is a little confusing. i'd probably remove some abstraction to help make it clearer what route we're transitioning to. i.e.

```javascript
navigate(pageFilter) {
  const route = `${this.mountPoint}.roles.index`;
  const args = [
    route,
    {
      queryParams: {
        pageFilter: pageFilter || null,
      },
    },
  ];
  this.router.transitionTo(...args);
}
```

Reply:
Good idea.